resolwe_bio/processes/expression/featureCounts.py

```diff
@@ -0,0 +1,888 @@
+        ]
+
+        if inputs.general.count_features:
+            args.append("-f")
+
+        if inputs.overlap.allow_multi_overlap:
+            args.append("-O")
+
+        if inputs.overlap.largest_overlap:
+            args.append("--largestOverlap")
+
+        if inputs.overlap.read_to_pos:
+            args.append("--read2pos")
+
+        if inputs.general.count_multi_mapping_reads:
+            args.append("-M")
+
+        if inputs.general.fraction:
+            args.append("--fraction")
+
+        if inputs.read_filtering.split_only:
+            args.append("--countSplitAlignmentsOnl")
+
+        if inputs.read_filtering.non_split_only:
+            args.append("--countNonSplitAlignmentsOnly")
+
+        if inputs.read_filtering.primary:
+            args.append("--primary")
+
+        if inputs.read_filtering.ignore_dup:
+            args.append("--ignoreDup")
+
+        if inputs.exon_exon_junctions.junc_counts:
+            args.append("-J")
+
+        if inputs.exon_exon_junctions.junc_counts and inputs.exon_exon_junctions.genome:
+            args.extend(["-G", inputs.exon_exon_junctions.genome.path])
+
+        if inputs.general.by_read_group:
+            args.append("--byReadGroup")
+
+        if inputs.general.count_long_reads:
+            args.append("-L")
+
+        if inputs.miscellaneous.report_reads:
+            args.append("-R CORE")
+
+        if inputs.miscellaneous.verbose:
+            args.append("--verbose")
+
+        # for paired-end samples, extend the args with paired-end input options
+        if paired_end and inputs.paired_end.is_paired_end:
+            args.append("-p")
+
+        if paired_end and inputs.paired_end.require_both_ends_mapped:
+            args.append("-B")
+
+        if paired_end and inputs.paired_end.check_frag_length:
+            args.append("-P")
+
+        if paired_end and inputs.paired_end.check_frag_length:
+            args.extend(["-d", inputs.paired_end.min_frag_length])
+
+        if paired_end and inputs.paired_end.check_frag_length:
+            args.extend(["-D", inputs.paired_end.max_frag_length])
+
+        if paired_end and inputs.paired_end.do_not_count_chimeric_fragments:
+            args.append("-C")
+
+        if paired_end and inputs.paired_end.do_not_sort:
+            args.append("--do_not_sort")
+
+        # Run featureCounts
+        return_code, _, _ = Cmd["featureCounts"][args][bam_file] & TEE(retcode=None)
+        if return_code:
+            self.error("Error while running featureCounts.")
+
+        # parse featureCounts output
+        get_gene_counts("featureCounts_rc.txt", "rc.txt", bam_file)
+        get_gene_lenghts("featureCounts_rc.txt", "gene_lengths.txt")
+
+        # Normalize counts
+        rnanorm_args = [
+            "rc.txt",
+            "--gene-lengths",
+            "gene_lengths.txt",
+            "--tpm-output",
+            "tpm.txt",
+            "--cpm-output",
+            "cpm.txt",
```

Comment:
This could go into a variable and be used throughout the script. IMO this would make it more resilient for future changes.